src/emmy/mechanics/hamilton.cljc

```diff
@@ -442,11 +441,12 @@
   "p.326"
   [C]
   (fn [s]
-    ((- J-func
-        (f/compose (Phi ((D C) s))
-                   J-func
-                   (Phi* ((D C) s))))
-     (s/compatible-shape s))))
+    (let [s-syms (s/compatible-shape s)]
```

Comment:
this forces `compatible-shape` to evaluate and make its gensym calls before the `((D C) s)` calls, important for repeatable tests